pennylane/hf/molecule.py

```diff
@@ -0,0 +1,153 @@
+        mult (int): Spin multiplicity :math:`\mathrm{mult}=N_\mathrm{unpaired} + 1` for
+            :math:`N_\mathrm{unpaired}` unpaired electrons occupying the HF orbitals. Possible
+            values of ``mult`` are :math:`1, 2, 3, \ldots`. If not specified, a closed-shell HF
+            state is assumed.
+        basis_name(str): Atomic basis set used to represent the molecular orbitals.
+        active_electrons (int): Number of active electrons. If not specified, all electrons are
+            considered to be active.
+        active_orbitals (int): Number of active orbitals. If not specified, all orbitals are
+            considered to be active.
+        l (tuple[int]): angular momentum numbers of the basis function.
+        alpha (array(float)): exponents of the primitive Gaussian functions
+        coeff (array(float)): coefficients of the contracted Gaussian functions
+        r (array(float)): positions of the Gaussian functions
+    """
+
+    def __init__(
+        self,
+        symbols,
+        coordinates,
+        charge=0,
+        mult=1,
+        basis_name="sto-3g",
+        active_electrons=None,
+        active_orbitals=None,
+        l=None,
+        alpha=None,
+        coeff=None,
+        r=None,
+    ):
+
+        if basis_name not in ["sto-3g", "STO-3G"]:
+            raise ValueError("The only supported basis set is 'sto-3g'.")
+
+        if set(symbols) - set(atomic_numbers):
+            raise ValueError(f"Atoms in {set(symbols) - set(atomic_numbers)} are not supported.")
+
+        self.symbols = symbols
+        self.coordinates = coordinates
+        self.charge = charge
+        self.mult = mult
+        self.basis_name = basis_name
+
+        self.n_basis, self.basis_data = mol_basis_data(self.basis_name, self.symbols)
+
+        if l is None:
+            l = [i[0] for i in self.basis_data]
+
+        if alpha is None:
+            alpha = [pnp.array(i[1], requires_grad=False) for i in self.basis_data]
+
+        if coeff is None:
+            coeff = [pnp.array(i[2], requires_grad=False) for i in self.basis_data]
+
+        if r is None:
+            r = sum([[self.coordinates[i]] * self.n_basis[i] for i in range(len(self.n_basis))], [])
+
+        self.l = l
+        self.alpha = alpha
+        self.coeff = coeff
+        self.r = r
+
+        self.basis_set = generate_basis_set(self.l, self.alpha, self.coeff, self.r)
+        self.n_orbitals = len(self.l)
+
+        self.nuclear_charges = generate_nuclear_charges(self.symbols)
+
+        self.n_electrons = sum(np.array(self.nuclear_charges)) - self.charge
+        self.core, self.active = qml.qchem.active_space(
+            self.n_electrons,
+            self.n_orbitals,
+            active_electrons=active_electrons,
+            active_orbitals=active_orbitals,
+        )
+
+
+def generate_basis_set(l, alpha, coeff, r):
+    r"""Generate a set of basis function objects.
```

Comment:
This is maybe not so descriptive, perhaps worth expanding to very succinctly explain what this function creates?